sway-fmt-v2/src/items/item_struct.rs

```diff
@@ -1,8 +1,210 @@
-use crate::fmt::{Format, FormattedCode, Formatter};
+use crate::{
+    config::items::ItemBraceStyle,
+    fmt::{Format, FormattedCode, Formatter},
+    utils::{
+        bracket::{AngleBracket, CurlyBrace},
+        item_len::ItemLen,
+    },
+};
 use sway_parse::ItemStruct;
+use sway_types::Spanned;
 
 impl Format for ItemStruct {
-    fn format(&self, _formatter: &mut Formatter) -> FormattedCode {
-        todo!()
+    fn format(&self, formatter: &mut Formatter) -> FormattedCode {
+        // TODO: creating this formatted_code with FormattedCode::new() will likely cause lots of
+        // reallocations maybe we can explore how we can do this, starting with with_capacity may help.
+        let mut formatted_code = FormattedCode::new();
+
+        // Get the unformatted
+
+        // Get struct_variant_align_threshold from config.
+        let struct_variant_align_threshold =
+            formatter.config.structures.struct_field_align_threshold;
+
+        // Should small structs formatted into a single line.
+        let struct_lit_single_line = formatter.config.structures.struct_lit_single_line;
+
+        // Get the width limit of a struct to be formatted into single line if struct_lit_single_line is true.
+        let config_whitespace = formatter.config.whitespace;
+        let width_heuristics = formatter
+            .config
+            .heuristics
+            .heuristics_pref
+            .to_width_heuristics(&config_whitespace);
+        let struct_lit_width = width_heuristics.struct_lit_width;
+
+        let multiline = !struct_lit_single_line || self.get_formatted_len() > struct_lit_width;
+        format_struct(
+            self,
+            &mut formatted_code,
+            formatter,
+            multiline,
+            struct_variant_align_threshold,
+        );
+        formatted_code
+    }
+}
+
+/// Format the struct if the multiline is passed as false struct will be formatted into a single line.
+///
+/// Example (multiline : false):
+/// struct Foo { bar: u64,  baz: bool }
+///
+/// Example (multiline : true):
+/// struct Foo {
+///  bar: u64,
+///  baz: bool,
+/// }
```

Comment:
When writing doc comments, try to be considerate of the markdown formatting, even if it's just for private functions. I believe they can show up in an LSP-enabled editor with markdown formatting when hovered, and they can also be generated using `cargo doc --document-private-items` which I personally find handy when familiarising with a new code-base.

Here's how the current formatting looks:

![Screenshot from 2022-06-23 11-22-53](https://user-images.githubusercontent.com/4587373/175187378-67ddf7ce-fb82-40a6-851c-2f1e07ea8c10.png)

Here's how it can look with some slight reformatting:

![Screenshot from 2022-06-23 11-27-27](https://user-images.githubusercontent.com/4587373/175187867-b7336d19-c370-4ccb-8a9c-38fe9d14db78.png)

```rust
/// Format the struct if the multiline is passed as false struct will be formatted into a single line.
///
/// ## Examples
///
/// (multiline : false):
///
/// ```rust,ignore
/// struct Foo { bar: u64,  baz: bool }
/// ```
///
/// (multiline : true):
///
/// ```rust,ignore
/// struct Foo {
///     bar: u64,
///     baz: bool,
/// }
/// ```
```

Reply:
Oh, that is really handy thanks. I will be using that to ensure that docs seem as I intended 👍

Reply:
@kayagokalp did you want to add this, or should I go ahead and approve as is?

Reply:
@eureka-cpu After merging this I will be completing https://github.com/FuelLabs/sway/pull/2097. So I was planning to fix the doc there so that mitch won't need to approve again but if you prefer the other way we can go that way too.